internal/pkg/uki/generate.go

```diff
@@ -183,6 +185,87 @@ func (builder *Builder) generatePCRPublicKey() error {
 	return nil
 }
 
+func (builder *Builder) generateProfiles() error {
+	defaultProfile := Profile{
```

Comment:
can we make this more generic e.g. by doing something like

```suggestion
for _, profile := range []struct{
  name string
  label string
  extraArgs []string
}{
{
   name: "main",
},
{
   name: "reset",
   label: "....",
   extraArgs: []string{....},
 }
}
	
```

Reply:
I wanted imager to be able to pass in extra profile, I can make this private though

Reply:
we can refactor it to be public structure, it's all good, but at least we DRY this piece a bit